sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -178,6 +178,8 @@ private[hive] class HiveClientImpl(
          """.stripMargin)
       hiveConf.set(k, v)
     }
+    // Disable CBO because we removed the Calcite dependency.
+    hiveConf.setBoolean("hive.cbo.enable", false)
```

Comment:
interesting. this doesn't affect spark since it's not reading this at all?

Reply:
This change works at least for Hive 1.x and Hive 2.x. 
for Hive 3.1, maybe add `calcite-core` is enough. Because since HIVE-14496, Hive.java uses `calcite-core`.

It is necessary to disable `hive.cbo.enable` here. You can verify it by:
```shell
build/sbt "hive/testOnly *.SQLQuerySuite *.HiveExternalCatalogSuite *.HiveDDLSuite *.HiveOrcSourceSuite *.HiveShowCreateTableSuite  *.DataSourceWithHiveMetastoreCatalogSuite" -Phive
```